lib/adagrams.rb

```diff
@@ -0,0 +1,166 @@
+# Chantelle and Semret (Pair-Programing)
+# Adagram project
+# 8/13/18
+require 'csv'
+
+# WAVE 1 SYNTAX
+def draw_letters
+
+  random_draw = []
+
+  letter_pool = ('A'..'Z').map do |char|
+    case char
+    when 'A', 'I'
+      letter = char * 9
+    when 'B', 'C', 'F', 'H', 'M', 'P', 'V', 'W', 'Y'
+      letter = char * 2
+    when 'D', 'L', 'S', 'U'
+      letter = char * 4
+    when 'E'
+      letter = char * 12
+    when 'G'
+      letter = char * 3
+    when 'J', 'K', 'Q', 'X', 'Z'
+      letter = char * 1
+    when 'N', 'R', 'T'
+      letter = char * 6
+    when 'O'
+      letter = char * 8
+    end
+  end
+
+  split_letter_pool = letter_pool.map do |chars|
+    chars.split("")
+  end
+
+  #random draw of 10 from array and flatten the letter pool
+  random_draw = split_letter_pool.flatten.sample(10)
+  return random_draw
+
+end
+
+# puts "Random draw: #{draw_letters}"
+
+def uses_available_letters?(word, letter_in_hand)
+  temp_hand = letter_in_hand.dup
+  inputted_word = []
+  inputted_word << word.upcase.split("")
+  inputted_word.flatten!
+
+  check_letter = true
+
+  inputted_word.each do |letter|
+    index = 0
+
+    if !temp_hand.include?(letter)
+      check_letter = false
+    end
+
+    if temp_hand.include?(letter)
+      index = temp_hand.index(letter)
+      temp_hand.delete_at(index)
+    end
+  end
+
+  return check_letter
+
+end
+
+# WAVE 3 SYNTAX
+# Scoring the word based on above parameters of scoring chart
+def score_word (word)
+
+  this_letter = []
+  this_letter << word.upcase.split("")
+  this_letter.flatten!
+
+  score = 0
+  this_letter.each do |value|
+
+    case value
+    when 'A', 'E', 'I', 'O' , 'U' , 'L' , 'N' , 'R' , 'S' , 'T'
+      score += 1
+    when 'D', 'G'
+      score += 2
+    when 'B', 'C', 'M', 'P'
+      score += 3
+    when 'F', 'H', 'V', 'W' , 'Y'
+      score += 4
+    when 'K'
+      score += 5
+    when 'J', 'X'
+      score += 8
+    when 'Q', 'Z'
+      score += 10
+    end
+  end
+
+
+  if word.length > 6 && word.length < 11
+    score += 8
+  end
+
+  return score
+end
+
+# WAVE 4 SYNTAX
+
+def highest_score_from(words)
+
+  # creating a hash to store highest scoring word with its score value
+  highest_scoring_hash = {
+    :score => 0,
+    :word => words.first
```

Comment:
You declare that the initial value of `highest_scoring_hash` has a value `0` for `score` and a value of the first word for `word`. You may want to make the initial value consistent and make the initial value of `score` equal to `score_word(words.first)`